content/en/blog/2021/stateful-set-made-easier/index.md

```diff
@@ -0,0 +1,226 @@
+tcp        0      0 127.0.0.1:2181          127.0.0.1:54560         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:54644         TIME_WAIT
+{{< /text >}}
+
+There are a few `ESTABLISHED` connections on both the 2888 and 3888 ports!  Next, let us check out the ZooKeeper server status.
+
+{{< text bash yaml >}}
+$ kubectl exec my-release-zookeeper-1 -c zookeeper -- /opt/bitnami/zookeeper/bin/zkServer.sh status
+/opt/bitnami/java/bin/java
+ZooKeeper JMX enabled by default
+Using config: /opt/bitnami/zookeeper/bin/../conf/zoo.cfg
+Client port found: 2181. Client address: localhost. Client SSL: false.
+Mode: follower
+{{< /text >}}
+
+It is exciting that the ZooKeeper service appears to be running this time! Let’s connect to each of the ZooKeeper pods from the `sleep` pod and run the command below to discover the server status of each pod within the ZooKeeper `StatefulSet`. Note, there is no need to deploy ServiceEntry resources for any of the ZooKeeper pods and we can call these pods directly using their DNS names (e.g. `my-release-zookeeper-0.my-release-zookeeper-headless`) from the `sleep` pod.
+
+{{< text bash yaml >}}
+$ kubectl exec -it deploy/sleep -c sleep -- sh  -c 'for x in my-release-zookeeper-0.my-release-zookeeper-headless my-release-zookeeper-1.my-release-zookeeper-headless my-release-zookeeper-2.my-release-zookeeper-headless; do echo $x; echo srvr|nc $x 2181; echo; done'
+my-release-zookeeper-0.my-release-zookeeper-headless
+Zookeeper version: 3.7.0-e3704b390a6697bfdf4b0bef79e3da7a4f6bac4b, built on 2021-03-17 09:46 UTC
+Latency min/avg/max: 1/7.5/20
+Received: 3845
+Sent: 3844
+Connections: 1
+Outstanding: 0
+Zxid: 0x200000002
+Mode: follower
+Node count: 6
+
+my-release-zookeeper-1.my-release-zookeeper-headless
+Zookeeper version: 3.7.0-e3704b390a6697bfdf4b0bef79e3da7a4f6bac4b, built on 2021-03-17 09:46 UTC
+Latency min/avg/max: 0/0.0/0
+Received: 3856
+Sent: 3855
+Connections: 1
+Outstanding: 0
+Zxid: 0x200000002
+Mode: follower
+Node count: 6
+
+my-release-zookeeper-2.my-release-zookeeper-headless
+Zookeeper version: 3.7.0-e3704b390a6697bfdf4b0bef79e3da7a4f6bac4b, built on 2021-03-17 09:46 UTC
+Latency min/avg/max: 0/0.0/0
+Received: 3855
+Sent: 3854
+Connections: 1
+Outstanding: 0
+Zxid: 0x200000002
+Mode: leader
+Node count: 6
+Proposal sizes last/min/max: 48/48/48
+{{< /text >}}
+
+Now our ZooKeeper service is running, let’s use Istio to secure all communication to our ZooKeeper and ZooKeeper headless services. Apply mutual TLS to the `default` namespace:
+
+{{< text bash >}}
+$ kubectl apply -n default -f - <<EOF
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
+metadata:
+  name: "default"
+spec:
+  mtls:
+    mode: STRICT
+EOF
+{{< /text >}}
+
+Continue sending some traffic from the `sleep` pod and bring up the Kiali dashboard to visualize the services in the `default` namespace:
+
+{{< image link="./view-zookeeper-from-kiali.png" caption="Visualize the ZooKeeper Services in Kiali" >}}
```

Comment:
describe what you're seeing here?

"The padlock icons on the traffic flows indicate that the connections are secure."

Reply:
Thanks, added.